commit a7ebbbb159c181c696770feeb89bf0334aaff6d8
Author: Wei Yongjun <weiyongjun1@huawei.com>
Date:   Wed Aug 17 08:03:29 2022 +0000

    fault-injection: allow stacktrace filter for x86-64
    
    This patchset allow fault injection to run on x86_64 and makes stacktrace
    filter work as expected.  With this, we can test a device driver module
    with fault injection more easily.
    
    
    This patch (of 4):
    
    FAULT_INJECTION_STACKTRACE_FILTER option was apparently disallowed on
    x86_64 because of problems with the stack unwinder:
    
        commit 6d690dcac92a84f98fd774862628ff871b713660
        Author: Akinobu Mita <akinobu.mita@gmail.com>
        Date:   Sat May 12 10:36:53 2007 -0700
    
            fault injection: disable stacktrace filter for x86-64
    
    However, there is no problems whatsoever with this today. Let's allow
    it again.
    
    Link: https://lkml.kernel.org/r/20220817080332.1052710-1-weiyongjun1@huawei.com
    Link: https://lkml.kernel.org/r/20220817080332.1052710-2-weiyongjun1@huawei.com
    Signed-off-by: Wei Yongjun <weiyongjun1@huawei.com>
    Cc: Akinobu Mita <akinobu.mita@gmail.com>
    Cc: Nathan Chancellor <nathan@kernel.org>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Kees Cook <keescook@chromium.org>
    Cc: Nick Desaulniers <ndesaulniers@google.com>
    Cc: Josh Poimboeuf <jpoimboe@kernel.org>
    Cc: Dan Williams <dan.j.williams@intel.com>
    Cc: Miguel Ojeda <ojeda@kernel.org>
    Cc: Isabella Basso <isabbasso@riseup.net>
    Cc: Vlastimil Babka <vbabka@suse.cz>
    Cc: Rasmus Villemoes <linux@rasmusvillemoes.dk>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index a93c45ebc320..5d19278f53c6 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -1982,7 +1982,6 @@ config FAIL_SUNRPC
 config FAULT_INJECTION_STACKTRACE_FILTER
 	bool "stacktrace filter for fault-injection capabilities"
 	depends on FAULT_INJECTION_DEBUG_FS && STACKTRACE_SUPPORT
-	depends on !X86_64
 	select STACKTRACE
 	depends on FRAME_POINTER || MIPS || PPC || S390 || MICROBLAZE || ARM || ARC || X86
 	help